games-roguelike/dwarf-fortress/dwarf-fortress-0.47.05.ebuild

```diff
@@ -0,0 +1,88 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit flag-o-matic optfeature prefix toolchain-funcs
+
+MY_PV=$(ver_rs 0- _ "$(ver_cut 2-)")
+MY_PN=df
+MY_P=${MY_PN}_${MY_PV}
+
+DESCRIPTION="The deepest, most intricate simulation of a world that's ever been created."
+HOMEPAGE="https://www.bay12games.com/dwarves"
+SRC_URI="amd64? ( https://www.bay12games.com/dwarves/${MY_P}_linux.tar.bz2 )
+	x86? ( https://www.bay12games.com/dwarves/${MY_P}_linux32.tar.bz2 )"
+
+LICENSE="free-noncomm BSD BitstreamVera"
+SLOT="0"
+KEYWORDS="-* ~amd64 ~x86"
+IUSE="debug"
+
+RDEPEND="media-libs/glew:0
+	media-libs/libsdl[joystick,video]
+	media-libs/sdl-image[png]
+	media-libs/sdl-ttf
+	sys-libs/zlib
+	virtual/glu
+	x11-libs/gtk+:2"
+
+# Yup, libsndfile, openal and ncurses are only needed at compile-time; the code
+# dlopens them at runtime if requested.
+DEPEND="${RDEPEND}
+	media-libs/libsndfile
+	media-libs/openal
+	sys-libs/ncurses-compat:5[unicode]"
+BDEPEND="virtual/pkgconfig"
+
+S="${WORKDIR}/${MY_PN}_linux"
+
+gamesdir="/opt/${PN}"
+QA_PREBUILT="${gamesdir#/}/libs/Dwarf_Fortress"
+RESTRICT="strip"
+
+src_prepare() {
+	# for eapply_user
+	default
+
+	# We don't want to use any of the bundled libraries
+	rm -f libs/*.so* || die
+	# This file seems to be missing this import, resulting in a compile error.
+	sed -i -e '1i#include <cmath>' g_src/ttf_manager.cpp || die
+	# This ensures that our wrapper script properly respects ${EPREFIX}
+	hprefixify "${FILESDIR}/dwarf-fortress"
```

Comment:
This would modify file in /var/db/repos/gentoo/...
Perhaps copy it from there first

Reply:
It gets copied [later in `src_install`](https://github.com/gentoo/gentoo/pull/20534/files#diff-960d65f2ff8959ba0548e979278eee2a8505e175ddf459502f467be84fbdc506R70).

Would it be better/preferred to first copy, then hprefixify, then move to /bin?

Reply:
Consider what happens if you install the package twice.
Best case, it will add the prefix twice, and therefore fail at runtime. Worst case, it will simply fail to build because checksum of the file doesn't match anymore.
Or should these case be reversed?

Reply:
Ah, I understand your concern.

Isn't `${FILEDIR}` already a copied version from `/var/db/repos/gento...`? i.e. isn't it typically somewhere like `/tmp/portage/games-rogueline/${P}/files`?

that's been my experience at least, and my understanding of the PMS, but please correct me if I'm wrong.

Reply:
`/tmp/portage/games-rogueline/${P}/files` is often a simple symlink to it

Reply:
Dang never mind, I guess I was wrong about this. TIL!

Reply:
I have resolved this. Please move any further reviews to #20721